modules/packages/Image.chpl

```diff
@@ -0,0 +1,322 @@
+                             colorComponent(colorB, red)) & colorMask;
+      const greenv = linInterp(t,
+                               colorComponent(colorA, green),
+                               colorComponent(colorB, green)) & colorMask;
+      const bluev = linInterp(t,
+                              colorComponent(colorA, blue),
+                              colorComponent(colorB, blue)) & colorMask;
+
+      outPixel = (redv << colorOffset(red)) |
+                 (greenv << colorOffset(green)) |
+                 (bluev << colorOffset(blue));
+    }
+
+    return outPixels;
+  }
+
+
+  /*
+    Scale a 2D array of pixels by a given factor
+
+    :arg arr: the 2D array of pixels
+    :arg factor: the factor to scale by
+    :returns: a new array of pixels scaled by the factor
+  */
+  proc scale(arr: [?d], factor: int): []
+    where d.isRectangular() && d.rank == 2 {
+    var rows = d.dim(0).size,
+        cols = d.dim(1).size;
+
+    var outArr: [d.dim(0).lowBound..#rows*factor, d.dim(1).lowBound..#cols*factor] arr.eltType;
+
+    const outArrSpace = {d.dim(0).lowBound..by factor #rows,
+                         d.dim(1).lowBound..by factor #cols};
+    forall (arrIdx, outArrIdx) in zip(d, outArrSpace) {
+      outArr[outArrIdx[0]..#factor, outArrIdx[1]..#factor] = arr[(...arrIdx)];
+    }
+
+    return outArr;
+  }
+
+
+  private import Subprocess as sp;
+  /*
+  Represents a ffmpeg stream that frame data can be written to to produce a mp4.
+  */
+  record mediaPipe {
+    @chpldoc.nodoc
+    var imageName: string;
+    @chpldoc.nodoc
+    var imgType: imageType;
+    @chpldoc.nodoc
+    var finished = false;
+    @chpldoc.nodoc
+    var process: sp.subprocess(true);
+
+    proc init(image: string, imgType: imageType, framerate: int = 30, rateFactor: int = 1) throws {
+      this.imageName = image;
+      this.imgType = imgType;
+      init this;
+
+      const ffmpegStart = ["ffmpeg", "-y",
+                           "-f", "image2pipe",
+                           "-r", framerate:string];
+      const pixelFmtArgs;
+      select imgType {
+        when imageType.bmp do
+          pixelFmtArgs = ["-pix_fmt", "bgr24", "-c:v", "bmp"];
+        otherwise {
+          halt("Don't know how to write images of type: ", imgType);
+          pixelFmtArgs = [""]; // dummy value for split init
+        }
+      }
+      const ffmpegEnd = ["-i", "-",
+                         "-c:v", "libx264",
+                         "-crf", rateFactor:string,
+                         "-pix_fmt", "yuv420p",
+                         this.imageName];
+      const nArgs = ffmpegStart.domain.size + pixelFmtArgs.domain.size + ffmpegEnd.domain.size;
+      var args: [0..#nArgs] string;
+      args[0..#ffmpegStart.domain.size] = ffmpegStart;
+      args[ffmpegStart.domain.size..#pixelFmtArgs.domain.size] = pixelFmtArgs;
+      args[(ffmpegStart.domain.size + pixelFmtArgs.domain.size)..#ffmpegEnd.size] = ffmpegEnd;
```

Comment:
Can you just use lists or something to not do this weird array slicing stuff?

Reply:
Sadly, no. `Subprocess.spawn` only takes an array